### PR TITLE
adds support for student users and adds specs

### PR DESCRIPTION
**Pull request by lawhorkl**
with this, students should receive their specific message when they register an account either locally or through shibboleth.

also adds relevant specs

any current branch will need to rebase and regenerate the dummy app to acquire the migrations needed

### CODE REVIEW

**Comment by lawhorkl** [comment]
@crowesn if you dont have a mailer set up on your dev environment, you'll need to check inside this array. It's where rails stores delivered emails in dev: https://github.com/uclibs/curate/pull/41/files#diff-0ede29dcc5b0c005747a1226d640adbeR8

File: db/migrate/20150625153556_add_dept_and_status_to_user.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class AddDeptAndStatusToUser < ActiveRecord::Migration
+  def change
+    add_column :users, :ucstatus, :string
+    add_column :users, :ucdepartment, :string
```
Comment (crowesn) [inline db/migrate/20150625153556_add_dept_and_status_to_user.rb:4]: Why are we adding the :ucdepartment field? Seems unrelated?

Comment (lawhorkl) [inline db/migrate/20150625153556_add_dept_and_status_to_user.rb:4]: It was a part of the migration that I copied from Scholar. It's not used anywhere, if you think it should be removed I can do that.
